### PR TITLE
Implement Associations

### PR DESCRIPTION
**Pull request by Kamba56**
- Added Book class
- Added Rental class
- Implement the association between book, rental, and person
- Added a classroom class
- Implement association between classroom and student

### CODE REVIEW

File: book.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+class Book
```
Comment (Ismailco) [inline book.rb:1]: - [ ] a book has many rentals kindly set up the associations here so relationship between Person and Book using the intermediate class Rental
e.g:
```ruby
def add_rental(person, date)
   Rental.new(date, self, person)
end
```

File: rent.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
+class Rental
+  attr_reader :person, :book
+  attr_accessor :date
```
Comment (Ismailco) [inline rent.rb]: - [ ] all three variables should have reader and writer access kindly fix this by adding the person and book to attr_accessor instead

**Review by Ismailco — CHANGES_REQUESTED** [review]
### Hi @Kamba56, 

Good job so far 👍🏼 
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

✔️  No linter errors
✔️ Used Gitflow
✔️ Work documented professionally

## Required Changes ♻️
- [ ] Good work using the README template provided by microverse however you should customize it to adapt your project
- [ ] When setting the classroom for a student it should also add it to the classroom's students, you can set up this association on the Student class 
  e.g:
  ```ruby
  def classroom=(classroom)
      @classroom = classroom
      classroom.students.push(self) unless classroom.students.include?(self)
  end
  ```
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by t-yanick — APPROVED** [review]
Hi @Kamba56, 

### Great job !!!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  Good commit messages
:heavy_check_mark: Descriptive Readme file

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:  
Congratulations! 🎉 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

### N/A
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**